### ISSUE TITLE
warn-unused-ignores doesn't fire when ignoring multiple codes

### ISSUE THREAD

**Issue by spaghEddieDoyle** [python/mypy#11157]
**To Reproduce**

```python
# test.py 

a = object()

def f() -> int:
    return a[b'c'] # type: ignore[index, override]
```

**Expected Behavior**

`a` is not indexable but ignored. 
An error along the lines of `unused "type: ignore" comment` for `override`

**Actual Behavior**

```shell
% mypy --warn-unused-ignores test.py 
Success: no issues found in 1 source file
```

**Your Environment**

```shell
% mypy --version
mypy 0.910
```
